src/EditorFeatures/Core.Wpf/InlineErrors/InlineErrorTag.cs

```diff
@@ -0,0 +1,126 @@
+        {
+            var block = new TextBlock
+            {
+                FontFamily = format.Typeface.FontFamily,
+                FontSize = 0.75 * format.FontRenderingEmSize,
+                FontStyle = FontStyles.Normal,
+                Foreground = format.ForegroundBrush,
+                Padding = new Thickness(left: 2, top: 0, right: 2, bottom: 0),
+                VerticalAlignment = VerticalAlignment.Center,
+            };
+
+            var id = new Run(_diagnostic.Id);
+            var link = new Hyperlink(id)
+            {
+                NavigateUri = new Uri(_diagnostic.HelpLink)
+            };
+
+            link.RequestNavigate += HandleRequestNavigate;
+            block.Inlines.Add(link);
+            block.Inlines.Add(": " + _diagnostic.Message);
+            block.Measure(new Size(double.PositiveInfinity, double.PositiveInfinity));
+            block.Arrange(new Rect(block.DesiredSize));
+
+            var border = new Border
+            {
+                Background = format.BackgroundBrush,
+                Child = block,
+                CornerRadius = new CornerRadius(2),
+                // Highlighting lines are 2px buffer. So shift us up by one from the bottom so we feel centered between them.
+                Margin = new Thickness(0, top: 0, 0, bottom: 2),
+                Padding = new Thickness(2)
+            };
+
+            border.Measure(new Size(double.PositiveInfinity, double.PositiveInfinity));
+
+            view.ViewportWidthChanged += ViewportWidthChangedHandler;
+            // Need to set these properties to avoid unnecessary reformatting because some dependancy properties
+            // affect layout.
+            // TODO: Not sure if these are needed anymore since the errors are not intratextadornment tags
+            TextOptions.SetTextFormattingMode(border, TextOptions.GetTextFormattingMode(view.VisualElement));
+            TextOptions.SetTextHintingMode(border, TextOptions.GetTextHintingMode(view.VisualElement));
+            TextOptions.SetTextRenderingMode(border, TextOptions.GetTextRenderingMode(view.VisualElement));
+
+            Canvas.SetTop(border, bounds.Bounds.Bottom - border.DesiredSize.Height);
+            Canvas.SetLeft(border, view.ViewportWidth - border.DesiredSize.Width);
+
+            return new GraphicsResult(border,
+                () => view.ViewportWidthChanged -= ViewportWidthChangedHandler);
+
+            void ViewportWidthChangedHandler(object s, EventArgs e)
+            {
+                Canvas.SetLeft(border, view.ViewportWidth - border.DesiredSize.Width);
+            }
+        }
+
+        /// <summary>
+        /// Navigates to the requrest URL
+        /// </summary>
+        private void HandleRequestNavigate(object sender, RoutedEventArgs e)
+        {
+            var link = (Hyperlink)sender;
+            var uri = link.NavigateUri.ToString();
+            Process.Start(uri);
```

Comment:
@sharwell may know if this is the right way to handle all links. I don't know if this impacts links to other code within the solution for example